### DISCUSSION TITLE
What is the purpose of firmwares in rootfs.img (included in install.img) ?

### DISCUSSION THREAD

**Discussion by viance** [rhinstaller/anaconda#5724] [Q&A]
I'm trying to create a custom RHEL installer ISO that is based on rhel-8.10-x86_64-boot.iso and uses kikstart for the automatic installation process. I noticed that install.img contains /usr/lib/firmware which takes up a lot of space.

What is the purpose of the /usr/lib/firmware in install.img if ISO installer will be include a linux-firmware RPM package in BaseOS (e.g. linux-firmware-20230824-119.git0e048b06.el8_9.noarch.rpm)?

Note: I installed such an ISO with a recreated install.img without anything in /usr/lib/firmware, and there were no problems during installation and operation on a some virtual machine and a hardware server (HP). I want to be sure that this won't cause any problems in other cases.

**Comment by jkonecny12** [comment]
If I understand it correctly, it works in a way that until you need a new kernel firmware to be loaded because of some filesystem or something like that you might be fine because it's already in the memory from the initrd. However, if you for example don't load btrfs into memory because it wasn't requested during teh boot it will fail.

However, I'm not 100% sure how the firmware sharing between initrd and installation environment (I guess the same happens on installed system) works. Maybe grub2 developers will be able to clarify / correct me here?
